### PR TITLE
Add LogLevel as logging framework

### PR DESCRIPTION
**Pull request by alextremp**
This PR will:

* Add configurable instance of LogLevel by external config
* Update README with configurable options and LogLevel reference
* Add some log info and debug traces

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -222,6 +222,28 @@ In order to do that, just call:
 openAds.resetConnectors()
 ```
 
+## Logging
+
+In order to debug what's happening with the Ads Loading, OpenAds uses [LogLevel](https://github.com/pimterry/loglevel) as it's logging framework.
```
Comment (rafamarquezv) [inline README.md]: We should avoid contractions ('s) in a formal text:

"In order to debug what is happening with Ads Loading, OpenAds uses [LogLevel](https://github.com/pimterry/loglevel) as its logging framework."

File: README.md [hunk 2]
```diff
@@ -222,6 +222,28 @@ In order to do that, just call:
...
+Also, it's using the [LogLevel Plugin Prefix](https://github.com/kutuluk/loglevel-plugin-prefix) plugin to configure how the log messages will be displayed.
```
Comment (rafamarquezv) [inline README.md]: "Also, it is using [LogLevel Plugin Prefix](https://github.com/kutuluk/loglevel-plugin-prefix) to configure how log messages will be displayed."

File: README.md [hunk 3]
```diff
@@ -222,6 +222,28 @@ In order to do that, just call:
...
+
+To configure it, you can set this node into the **config** object when initializing OpenAds:
```
Comment (rafamarquezv) [inline README.md]: In order to set the Logger up, add 'LogLevel' node entry into the **config** object when calling OpenAds 'init' method as follows:

File: README.md (hunk 1 above)
Comment (victuxbb) [reply README.md]: :+1:

File: README.md (hunk 2 above)
Comment (victuxbb) [reply README.md]: :+1:

File: README.md (hunk 3 above)
Comment (victuxbb) [reply README.md]: :+1:

**Comment by victuxbb** [comment]
We have pending on this PR:
* Add an interface logger in domain for decouple domain and infrastructure
* Implement domain logger interface with LogLevel library
* Add feature to scan url and active debug mode (will require to have access to DOMDriver)
* Test coverage to this feature

**Comment by alextremp** [comment]
The last changes includes these requested points:
* Add an interface logger in domain for decouple domain and infrastructure
* Implement domain logger interface with LogLevel library

**Comment by victuxbb** [comment]
Good coverage!

File: src/test/openads/domain/service/LoggerInitializerTest.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-unused-expressions */
```
Comment (rafamarquezv) [inline src/test/openads/domain/service/LoggerInitializerTest.js]: This line should not be necessary anymore... ^_^

**Comment by alextremp** [comment]
Done @rafamarquezv

File: src/openads/infrastructure/logger/LogLevelLoggerFactory.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
 import LogLevelLogger from './LogLevelLogger'
+import LoggerFactory from '../../domain/logger/LoggerFactory'
 
-export default class LogLevelLoggerFactory {
+export default class LogLevelLoggerFactory extends LoggerFactory {
```
Comment (rafamarquezv) [inline src/openads/infrastructure/logger/LogLevelLoggerFactory.js]: Nailed!

File: src/test/openads/domain/logger/LoggerFactoryTest.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable no-unused-expressions */
```
Comment (rafamarquezv) [inline src/test/openads/domain/logger/LoggerFactoryTest.js]: unnecessary

File: src/openads/infrastructure/logger/LogLevelLoggerFactory.js (hunk 5 above)
Comment (alextremp) [reply src/openads/infrastructure/logger/LogLevelLoggerFactory.js]: I know it has been only to put a free meme ;)

File: README.md [hunk 7]
```diff
@@ -243,6 +243,8 @@ const openAds = OpenAds.init({config: {
 }})
 ```
 
+Navigating in a browser, adding **openads_debug** to the URL's query string, will enable the debug mode automatically
```
Comment (rafamarquezv) [inline README.md]: I would use this sentence: 

"While navigating in a browser, adding **openads_debug** keyword to the URL query string will enable the debug mode automatically."

Maybe an example would be helpful.

Comment (alextremp) [reply README.md]: Done

**Pull request merged**